src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -125,8 +125,8 @@ public static int Main(string[] args)
             log.Info("Initializing Services");
             engine.Initialize();
 
-            // Owns the channel used for communications with the agency and with clients
-            var testAgencyServer = engine.Services.GetService<TestAgency>();
+            // Make sure we can communicate with the agency
+            ServerUtilities.GetTcpChannel("", 0, 2);
```

Comment:
I'm tempted to put the code itself here and not call server utilities as it may confuse people and you end up checking anyway.